src/Features/LanguageServer/Protocol/ExternalAccess/Razor/GetProjectContextsHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Roslyn.Utilities;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.ExternalAccess.Razor;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(GetProjectContextsHandler)), Shared]
+[Method(GetProjectContextsName)]
+internal class GetProjectContextsHandler : ILspServiceDocumentRequestHandler<ProjectContextsParams, ProjectContextList?>
```

Comment:
@jasonmalinowski should review this.
One thought I had is do we need a separate endpoint?  We could potentially have Razor provide an additional parameter indicating whether it wants other project info - then return a custom subtype of the project information if that parameter is provided

Reply:
lol, Andrew and I had that exact conversation :)

Reply:
Yeah, that would seem logical. :smile: Let's chat more about this since I think for a project context picker in VS Code I might also need to add something more to our existing call, and it'd be good to just unify.

Reply:
Definitely happy to discuss and find a solution here. I did the 'this is what razor needs' pr because I don't fully know Roslyn plans. If you're going to need to add on more information then by all means we can benefit together.